### DISCUSSION TITLE
ValueError: The qform and sform matrices do not match.

### DISCUSSION THREAD

**Discussion by momo-1120** [simnibs/simnibs#469] [7.Creating head models]
Hello, I encountered an error while generating a head model with charm: "The qform and sform matrices do not match." I would like to know the reason for this and whether it is necessary to register both T1W and T2W to the MNI space for personalized tES simulation.
The data I used comes from the first type shown in the figure . Did I make a wrong choice?
<img width="337" alt="微信图片_20241108200250" src="https://github.com/user-attachments/assets/0dd41d38-8d11-4c36-9858-cf08818727e6">

**Comment by TorgeW** [comment]
You choose the correct data.
when running charm please use --forceqform, if that doesnt work use --forcesform.

**Reply by momo-1120** [reply]
Thank you for your reply. I noticed that the dataset I chose is raw data and has not undergone any preprocessing, such as bias field correction. Will this have an impact on the simulation?

**Reply by oulap** [reply]
charm will bias field correct the data as a part of the segmentation algorithm so you don't need to preprocess the data other than create a nifti file in case your data is dicom

**Reply by momo-1120** [reply]
However, I checked the segmented tissue and found that some parts of the tissue were missing.
![图片1](https://github.com/user-attachments/assets/a2fd038f-cbbf-424c-b648-79635ef63d77)

**Reply by oulap** [reply]
That looks like a sinus, which is a cavity inside the skull. You can double-check by overlaying the segmentation on your input scan, although the cavities are pretty hard to distinguish from compact bone on an MRI scan.
